offchain/danaswap-browser/src/DanaSwap/Browser/Home.purs

```diff
@@ -19,18 +24,239 @@ main = HA.runHalogenAff do
     Nothing -> throwError $ error "Cannot find #home"
     Just elem -> void $ runUI component unit elem
 
+type State =
+  { pools :: Array Pool
+  , currentPool :: Maybe Pool
```

Comment:
Can we end up in an invalid state here here the `currentPool` and `pools` are out of sync? If yes, I would suggest making `pools` some sort of `Map` and `currentPool` holding onto a `newtype ID = ID String`

Reply:
@toastal what the initial value for `currentPool` if it's a `ID`

Reply:
`currentPool ∷ Maybe ID` XD

Reply:
I reckon that has the same issue as you mentioned earlier, the `initialCurrentPool` and `Pools` might go async

Reply:
I reckon that has the same issue as you mentioned earlier, the `initialCurrentPool` and `Pools` might go out of async

Reply:
You wanna know what a good option would be? Port out [Choice](https://gitlab.com/platonic/shpadoinkle/-/blob/master/widgets/Shpadoinkle/Widgets/Types/Choice.hs) that @Fresheyeball wrote for [Shpadoinkle](https://shpadoinkle.org/) to PureScript. He recommended I do that a while ago, but I hadn't yet gotten around to needing it (and the `type family` told me it wasn't going to be a 1-to-1 trivial, thing)--and we do have the [Platonic.Systems GitLab org](https://gitlab.com/platonic) to publish stuff to.

`Choice` encapsulates the generic common UI pattern: I have collection of things and I can optionally choose one or many of those options. Since it has laws and such, it would be better than whatever ad hoc thing we're doing and would come in handy literally all over the place. I just wish there were more text documentation in that file to explain what its solving so I didn't have to fumble around explaining it.

Reply:
wow, I reckon this will beyond the scope of MVP :)

Reply:
But we really *should* consider something like this for long term. It's also just one file and could save us headaches now and soon. Open-sourcing it may be a bit out of scope though, sure XD